Source/Library/Adapt.Analytics/MagnitudeCalculationAnalytic.cs

```diff
@@ -0,0 +1,109 @@
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  08/02/2021 - C. Lackner
+//       Generated original version of source code.
+//  12/10/2021 - A. Hagemeyer
+//       Changed to a magnitude calculation analytic
+//
+// ******************************************************************************************************
+
+
+using Adapt.Models;
+using GemstoneCommon;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Numerics;
+using System.Runtime;
+using System.Threading.Tasks;
+using System.Windows;
+
+namespace Adapt.DataSources
+{
+    /// <summary>
+    /// Finds the imaginary component of the complex number given magnitude and phase
+    /// </summary>
+
+    [AnalyticSection(AnalyticSection.DataCleanup)]
+
+    [Description("Magnitude Calculation: Return magnitude of a complex number")]
+    public class MagnitudeCalculation : IAnalytic
+    {
+        private Setting m_settings;
+        private int m_fps;
+        public class Setting
+        {
+            public AngleUnit Unit { get; set;  }
+        }
+
+        public Type SettingType => typeof(Setting);
+
+        public int FramesPerSecond => m_fps;
+
+        int IAnalytic.PrevFrames => 0;
+
+        int IAnalytic.FutureFrames => 0;
+
+        public IEnumerable<string> OutputNames()
+        {
+            return new List<string>() { "Calculated Magnitude" };
+        }
+
+        public IEnumerable<string> InputNames()
+        {
+            return new List<string>() { "Magnitude", "Phase"};
+        }
+
+        public Task<ITimeSeriesValue[]> Run(IFrame frame, IFrame[] previousFrames, IFrame[] futureFrames)
+        {
+            AdaptValue result = new AdaptValue("Calculated Magnitude", getComplex(frame).Magnitude, frame.Timestamp);
+            return Task.FromResult<ITimeSeriesValue[]>(new AdaptValue[] { result });
+        }
+
+        public Complex getComplex(IFrame frame) 
+        {
+            ITimeSeriesValue magnitude = frame.Measurements["Magnitude"];
+            ITimeSeriesValue phase = frame.Measurements["Phase"];
+            if (m_settings.Unit == AngleUnit.Degrees)
+                return new Complex(magnitude.Value * Math.Cos(phase.Value), magnitude.Value * Math.Sin(phase.Value));
+            else 
+                return new Complex(magnitude.Value * Math.Cos((180 / Math.PI) * phase.Value), magnitude.Value * Math.Sin((180 / Math.PI) * phase.Value));
+            
+        }
+
+        public void Configure(IConfiguration config)
+        {
+            m_settings = new Setting();
+            config.Bind(m_settings);
+        }
+
+        public void SetInputFPS(IEnumerable<int> inputFramesPerSecond)
```

Comment:
For multi-input analytics use Least Common Deniominator; For single input just return `.FirstOrDefault()`